File: Práctica Grupal 2/practica2-grupo1-Equipo2/src/gestorAplicacion/clasesDeAdministracion/Multa.py
from datetime import date

class Multa:
    numero_multas = 0  # Variable de clase para llevar un registro del número de multas

    def __init__(self, tipo, fecha_impuesta, usuario, costo):
        self.id_multa = Multa.numero_multas
        self.tipo = tipo
        self.fecha_impuesta = fecha_impuesta
        self.usuario = usuario
        self.costo = costo
        Multa.numero_multas += 1

    def get_id_multa(self):
        return self.id_multa

    def get_tipo(self):
        return self.tipo

    def set_tipo(self, tipo):
        self.tipo = tipo

    def get_fecha_impuesta(self):
        return self.fecha_impuesta

    def get_usuario(self):
        return self.usuario
    
    def get_costo(self):
        return self.costo
    
    def pagar_multa(self):
        # Eliminar la multa del registro del usuario
        self.usuario.eliminar_multa(self)

    

File: Práctica Grupal 1/src/gestorAplicacion/paquete2/Multa.java
package gestorAplicacion.paquete2;

import java.util.Date;
import java.io.Serializable;

public class Multa implements Serializable{
	private static final long serialVersionUID = 1L;
    private int idMulta; // Identificador único de la multa
    private String tipo; // Tipo de la multa (por ejemplo, "Retraso en la devolución")
    private Date fechaImpuesta; // Fecha en la que se impuso la multa
    private Usuario usuario; // Usuario al que se impuso la multa
    private static int numeroMultas;

    // Constructor de la clase Multa
    public Multa(String tipo, Date fechaImpuesta, Usuario usuario) {
        this.idMulta = numeroMultas;
        this.tipo = tipo;
        this.fechaImpuesta = fechaImpuesta;
        this.usuario = usuario;
        setNumeroMultas(getNumeroMultas() + 1);
    }

    // Método para obtener el identificador único de la multa
    public int getIdMulta() {
        return idMulta;
    }

    // Método para obtener el tipo de la multa
    public String getTipo() {
        return tipo;
    }

    // Método para establecer el tipo de la multa
    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    // Método para obtener la fecha en la que se impuso la multa
    public Date getFechaImpuesta() {
        return fechaImpuesta;
    }

    // Método para obtener el usuario al que se impuso la multa
    public Usuario getUsuario() {
        return usuario;
    }

    // Método para pagar la multa
    public String pagarMulta() {
        // Eliminar la multa del registro del usuario
        usuario.eliminarMulta(this);
        // Notificar al usuario sobre el pago exitoso de la multa
        return("La multa de ID: "+idMulta+" con tipo '" + tipo + "' y Fecha " + fechaImpuesta + " ha sido pagada.");
    }

	public static int getNumeroMultas() {
		return numeroMultas;
	}

	public static void setNumeroMultas(int numeroMultas) {
		Multa.numeroMultas = numeroMultas;
	}

}
